documentation/design-docs/dotnet-monitor.yaml

```diff
@@ -0,0 +1,457 @@
+swagger: "2.0"
+info:
+  description: "dotnet-monitor diagnostics agent"
+  version: "1.0.0"
+  title: "dotnet-monitor"
+host: "localhost:9999"
+basePath: "/diag/v1/"
+schemes:
+- "https"
+- "http"
+paths:
+  /processes/:
+    get:
+      tags:
+      - "processes"
+      operationId: "getprocesses"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            type: array
+            items: 
+              $ref: "#/definitions/Process"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /dump/:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getdefaultprocessdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "type"
+        in: query
+        description: "Specifies the dump type, which determines the kinds of information that are collected from the process."
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /dump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getprocessdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process to collect a process dump"
+        required: true
+        type: "integer"
+      - name: "type"
+        in: query
+        description: "Specifies the dump type, which determines the kinds of information that are collected from the process."
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump/:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getdefaultgcdump"
+      produces:
+      - "application/octet-stream"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getgcdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
```

Comment:
you may need some sort of time/duration parameter for gc dumps, since it's not guaranteed to happen right away. This is a timeout paramter on the command line tool

Reply:
> This is a timeout parameter on the command line too

I don't believe so.

I thought @noahfalk's feedback was don't add stuff to the the APIs if doesn't do anything. Timeout is something we should in a future version of the API when we can propagate it all the way through

Reply:
Gc dumps appear to be collected by processing events from the event stream. You probably don't want to allow this to happen forever, but rather time constrain it, or end once it actually happens.

Reply:
Ah, you're right. I'll add a timeout